projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -772,8 +804,9 @@ export class DatagridComponent<R> extends CanTranslate(class {}) implements OnIn
     }
 
     ngAfterViewInit(): void {
-        setTimeout(() => {
+        this.viewInitted = true;
+        if (this.pagination.pageSize === 'Magic') {
```

Comment:
Let's add a line `this.cd.detectChanges()` with a comment explaining that for magic page sizes, we have no alternative but to modify the pageSize after the components has rendered, so we use this to avoid `ExpressionChangedAfterItHasBeenCheckedError` or even worse, a `setTimeout`